### PR TITLE
Spring + camel init

### PR DESCRIPTION
**Pull request by wojnarfilip**
Initiating spring and camel project (first route + initial route test with restassured)

### CODE REVIEW

File: src/main/java/patriot/virtualsmarthomeplus/InitialApplication.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package patriot.virtualsmarthomeplus;
```
Comment (tondrusk) [inline src/main/java/patriot/virtualsmarthomeplus/InitialApplication.java:1]: Use proper packaging - `io.patriot_framework.virtualsmarthomeplus`

File: src/main/java/patriot/virtualsmarthomeplus/InitialApplication.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+
+@SpringBootApplication
+public class InitialApplication {
+
+	public static void main(String[] args) {
+		SpringApplication.run(InitialApplication.class, args);
+	}
+
```
Comment (tondrusk) [inline src/main/java/patriot/virtualsmarthomeplus/InitialApplication.java:12]: Inconsistency in files. I prefer **NOT** leaving a blank line here.

File: src/main/java/patriot/virtualsmarthomeplus/house/HouseBean.java [hunk 3]
```diff
@@ -0,0 +1,14 @@
+package patriot.virtualsmarthomeplus.house;
+
+public class HouseBean {
+    private House house;
```
Comment (tondrusk) [inline src/main/java/patriot/virtualsmarthomeplus/house/HouseBean.java:4]: Again, inconsistency. I would leave a blank line after the class declaration.

File: src/main/java/patriot/virtualsmarthomeplus/house/HouseBean.java [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+    private HouseBean(House house) {
+        this.house = house;
+    }
+
+    public void fireplaceOn() {
+        ((Fireplace) (house.getDevice("fireplace"))).fireUp();
+    }
```
Comment (tondrusk) [inline src/main/java/patriot/virtualsmarthomeplus/house/HouseBean.java:12]: Dead method ☠️

File: src/main/java/patriot/virtualsmarthomeplus/house/devices/Actuator.java [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+    private boolean enabled = false;
+
+    public Actuator(String label) {
+        super(label);
+    }
+
+    @Override
+    public void switchIt(boolean on) {
+        this.enabled = on;
```
Comment (tondrusk) [inline src/main/java/patriot/virtualsmarthomeplus/house/devices/Actuator.java:13]: Redundant `this` call _(preferences)_.

File: src/main/java/patriot/virtualsmarthomeplus/house/House.java [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+import patriot.virtualsmarthomeplus.house.devices.Device;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public final class House {
+
+    private String houseName;
+    private Map<String, Device> devices;
```
Comment (tondrusk) [inline src/main/java/patriot/virtualsmarthomeplus/house/House.java:12]: ⚠️ **NPE**

Comment (tondrusk) [reply src/main/java/patriot/virtualsmarthomeplus/house/House.java:12]: Should be initialized with an empty Map.

File: src/test/java/patriot/virtualsmarthomeplus/JSONTests.java [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+    }
+
+    @Test
+    void AcceptingJSONPost() {
+        given()
+                .contentType(ContentType.JSON)
+                .port(8080)
+                .when().post("http://localhost:8080/fireplace")
+                .then().statusCode(200);
+    }
```
Comment (tondrusk) [inline src/test/java/patriot/virtualsmarthomeplus/JSONTests.java:49]: 🟥 RED tests

File: src/main/java/patriot/virtualsmarthomeplus/house/House.java [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+
+    public House(String houseName) {
+        this.houseName = houseName;
+    }
+
+    public void addDevice(String label, Device device) {
+        devices.put(label, device);
+    }
```
Comment (tondrusk) [inline src/main/java/patriot/virtualsmarthomeplus/house/House.java:20]: ⚠️ **NOT** covered with tests

File: src/main/java/patriot/virtualsmarthomeplus/house/House.java [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+    public <T> List<T> getAllDevices(Class<T> type) {
+        ArrayList<T> toReturn = new ArrayList<>();
+
+        for (Device d : devices.values()) {
+            if (type.isInstance(d)) {
+                toReturn.add(type.cast(d));
+            }
+        }
+        return toReturn;
+    }
```
Comment (tondrusk) [inline src/main/java/patriot/virtualsmarthomeplus/house/House.java:51]: ⚠️ **NOT** covered with tests

File: src/main/java/patriot/virtualsmarthomeplus/house/Fireplace.java [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+        super(label);
+    }
+
+    public void fireUp() {
+        this.switchIt(true);
+    }
+
+    public void extinguish() {
+        this.switchIt(false);
+    }
```
Comment (tondrusk) [inline src/main/java/patriot/virtualsmarthomeplus/house/Fireplace.java:17]: Uncovered

**Review by mijaros — APPROVED** [review]

**Review by tondrusk — APPROVED** [review]

**Pull request merged**